.github/workflows/semgrep.yml

```
@@ -18,5 +18,5 @@ jobs:
     container:
       image: returntocorp/semgrep@sha256:0f12ff85e0e3d7e586190196c723e2fc47c4f3d99c760c37a74c1ee1fffb7bff
     steps:
-    - uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3
+    - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4
```

```suggestion
    - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
```

Evidently, this PR is for https://github.com/actions/checkout/releases/tag/v4.1.1 (just compare the commit hash of the release commit).

---

i have raised this in https://github.com/OpenINF/.github/issues/184#issuecomment-1862966945